vispy/scene/widgets/widget.py

```diff
@@ -64,8 +64,11 @@ def __init__(self, pos=(0, 0), size=(10, 10), border_color=None,
         
         # layout interaction
         # todo: use Cassowary; see #277 
-        self._fixed_size = (None, None)
+        self._width_limits = [0, None]
+        self._height_limits = [0, None]
         self._stretch = (None, None)
+        self.var_w = self.var_h = None  # HACK
```

Comment:
Like the line says, hack :) I was messing about with the constraint solving when I was trying this out. I'll rebase and squash these commits so it doesn't show up in history